mne/io/eeglab/eeglab.py

```diff
@@ -260,6 +327,247 @@ def read_epochs_eeglab(input_fname, events=None, event_id=None, montage=None,
+    """Returns list of Bunch objects - A Bunch object is a
+       dictionary-like object that exposes its keys as attributes.
+       ASSUMES: The group consists solely of arrays of HDF5 obj refs,
+       and that these refs all reference 2D numpy arrays that need to
+       be flattened to either 1D arrays or Scalars"""
+
+    import h5py
+
+    try:
+        # h5_values gives dict of 1D arrays of HDF obj references
+        h5_values = {ct: in_hlGroup[ct].value.flatten() for ct in in_hlGroup}
+
+        # derefs dereferences HDF obj references and cnverts arrays with
+        # shapes = (1,) to scalars. Returns adict mapping keys to lists
+        # of arrays and scalars
+        derefs = {x: [orig[y].value.flatten()
+                      if orig[y].value.flatten().shape != (1,)
+                      else orig[y].value.flatten()[0]
+                      for y in h5_values[x]]
+                  if isinstance(h5_values[x], np.ndarray) and
+                  isinstance(h5_values[x][0], h5py.Reference)
+                  else h5_values[x]
+                  for x in h5_values}
+
+        for ct in in_hlGroup:
+            msg = indent + "Converting " + tuple_name + '_' + ct
+            logger.debug(msg)
+
+    except IOError:
+        derefs = {ct: [None] for ct in in_hlGroup}
+        warn("Couldn't read", tuple_name, ". Assuming empty")
+
+    sz = len(derefs[list(derefs.keys())[0]])
+    bch_list = [Bunch(**{key: derefs[key][x] for key in derefs})
+                for x in range(sz)]
+    return bch_list
+
+
+def _get_eeg_data(input_fname, uint16_codec=None):
+    from scipy import io
+    try:
+        # Try to read old style Matlab file
+        fhandle = open(input_fname, 'rb')
+        eeg = io.loadmat(fhandle, struct_as_record=False,
+                         squeeze_me=True,
+                         uint16_codec=uint16_codec)['EEG']
+        print("Closing    %s" % input_fname)
+        fhandle.close()
+    except NotImplementedError:
+        # Try to read new style Matlab file (Version 7.3+)
+        # Note: Now eeg will be returned as a Bunch object,
+        # instead of an io.matlab.mio5_params.mat_struct object.
+        fhandle.close()
+        import h5py  # Added to read newer Matlab files (7.3 and later)
+        logger.info("Attempting to read Matlab style hdf file")
+        f = h5py.File(input_fname)
+        eeg_dict = hdf_2_dict(f, f['EEG'], parent=None)
+        f.close()
+        eeg = _bunchify(eeg_dict)
+
+        str_conversion_fields = ('datfile', 'filename', 'filepath',
+                                 'history', 'ref', 'saved', 'setname')
+        for curr_field in str_conversion_fields:
+            temp = eeg[curr_field]
+            c1 = isinstance(temp, np.ndarray)
+            c2 = c1 and np.issubdtype(temp.dtype, np.integer)
+            c3 = c2 and (min(temp) >= 0)
+            if c3:
+                eeg[curr_field] = ''.join([chr(y) for y in temp])
+
+    return eeg
```

Comment:
This function, which does the triaging and a tiny bit of custom conversion should stay here.

Make sense?